var/spack/repos/builtin/packages/py-datalad/package.py

```diff
@@ -97,24 +106,31 @@ class PyDatalad(PythonPackage):
         depends_on("py-vcrpy", type=("build", "run"))
         depends_on("py-nose@1.3.4:", when="@:0.16", type=("build", "run"))
 
+    with when("+duecredit"):
+        depends_on("py-duecredit", type=("build", "run"))
+
+    # for version @:0.17
+    with when("+metadata"):
+        depends_on("py-simplejson", type=("build", "run"))
+        depends_on("py-whoosh", type=("build", "run"))
+
+    # for version @:0.17
     with when("+metadata-extra"):
         depends_on("py-pyyaml", type=("build", "run"))
         depends_on("py-mutagen@1.36:", type=("build", "run"))
         depends_on("py-exifread", type=("build", "run"))
         depends_on("py-python-xmp-toolkit", type=("build", "run"))
         depends_on("pil", type=("build", "run"))
 
-    with when("+duecredit"):
-        depends_on("py-duecredit", type=("build", "run"))
-
     # full
     # use conflict to avoid to have to maintain the dependencies twice
     conflicts("~downloaders-extra", when="+full")
     conflicts("~misc", when="+full")
     conflicts("~tests", when="+full")
-    conflicts("~metadata-extra", when="+full")
     conflicts("~duecredit", when="+full")
 
+    conflicts("~metadata-extra", when="+full @:0.17")
```

Comment:
I'm not sure if this version is needed. In versions 0.18+, the metadata-extra variant doesn't even exist, so it can't be + or ~. Can you remove it and test?

Reply:
I was not sure how this would be handled, that's why I added it to be on the safe side. After some tests, it seems to work correctly also without the version restriction: version 0.17.5+full has the dependencies, version 18.3+full not.
So I removed it.